### PR TITLE
ci: use reuseable workflows

### PR DESCRIPTION
**Pull request by huitseeker**

### CODE REVIEW

**Comment by huitseeker** [comment]
Hmm, I forgot to remove the part of the test that runs the doctests, which makes the failure of https://github.com/lurk-lab/neptune/actions/runs/7294189991/job/19878612122?pr=255 weird.

**Comment by huitseeker** [comment]
This is the line that makes this tricky, reduced the PR to something simpler for now:
https://github.com/lurk-lab/neptune/blob/main/gbench/Cargo.toml#L25

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -101,6 +93,10 @@ jobs:
       NVIDIA_DRIVER_CAPABILITITES: compute,utility
       EC_GPU_FRAMEWORK: cuda
     steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: lurk-lab/ci-workflows
+      - uses: ./.github/actions/ci-env
```
Comment (samuelburnham) [inline .github/workflows/rust.yml:99]: We can also use the `gpu-setup` action to cut down on more lines in both the `cuda` and `opencl` jobs

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -92,6 +74,16 @@ jobs:
     - name: Check Rust MSRV
       run: cargo msrv verify
 
+  # Check documentation links aren't broken
+  link-checker:
+    uses: lurk-lab/ci-workflows/.github/workflows/links-check.yml@main
+    with: 
+      fail-fast: true
+
+  # Lint dependencies for licensing and auditing issues as per https://github.com/lurk-lab/arecibo/blob/main/deny.toml
```
Comment (samuelburnham) [inline .github/workflows/rust.yml:83]:
```suggestion
  # Lint dependencies for licensing and auditing issues as per https://github.com/lurk-lab/neptune/blob/main/deny.toml
```

**Review by samuelburnham — CHANGES_REQUESTED** [review]

**Comment by huitseeker** [comment]
Pausing work on this: we need to switch to https://github.com/dtolnay/rust-toolchain first, before we use our reusable workflows.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (huitseeker) [reply .github/workflows/rust.yml:99]: That's definitely the next step!

**Review by samuelburnham — APPROVED** [review]

**Pull request merged**